fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java

```diff
@@ -449,6 +506,11 @@ public PlanNode getPlanRoot() {
 
     public void setPlanRoot(PlanNode root) {
         planRoot = root;
+        if (root instanceof JoinNode) {
```

Comment:
This implementation should have bug, you should check the children also.

Reply:
bugless, when every PlanNode added to a PlanFragment during PlanFragment building, setPlanRoot is called.